design/authenticators/authn_k8s/inject_client_cert_solution_design.md

```diff
@@ -0,0 +1,255 @@
+# Solution Design - Inject Client Cert fix
+[//]: # "Change the title above from 'Template' to your design's title"
+
+## Table of Contents
+- [Solution Design - Inject Client Cert fix](#solution-design---inject-client-cert-fix)
+  * [Useful links](#useful-links)
+  * [Background](#background)
+  * [Issue description](#issue-description)
+  * [Solution](#solution)
+    + [Waiting for the injection to finish](#waiting-for-the-injection-to-finish)
+    + [Inject_client_cert response code](#inject-client-cert-response-code)
+    + [Cert injection implementation](#cert-injection-implementation)
+      - [Option 1 - Run a parallel request to verify cert injection](#option-1---run-a-parallel-request-to-verify-cert-injection)
+      - [Option 2 - Send the cert content as part of the URL](#option-2---send-the-cert-content-as-part-of-the-url)
+      - [Option 3 - Client sets its State once cert is injected](#option-3---client-sets-its-state-once-cert-is-injected)
+      - [Option 4 - Send cert as part of bash script](#option-4---send-cert-as-part-of-bash-script)
+      - [Conclusion](#conclusion)
+    + [Design](#design)
+    + [Backwards compatibility](#backwards-compatibility)
+    + [Performance](#performance)
+    + [Security](#security)
+  * [Affected Components](#affected-components)
+  * [Cross-team dependencies](#cross-team-dependencies)
+  * [Test Plan](#test-plan)
+  * [Logs](#logs)
+    + [Audit](#audit)
+  * [Documentation](#documentation)
+  * [Delivery plan](#delivery-plan)
+    + [Total](#total)
+    + [Parallelism](#parallelism)
+    + [Spikes and researches](#spikes-and-researches)
+    + [Implementation](#implementation)
+    + [Testing](#testing)
+    + [Demo](#demo)
+    
+## Useful links
+[//]: # "Add links that may be useful for the reader"
+
+- [Known issue in k8s exec](https://github.com/kubernetes/kubernetes/issues/89899)
+
+## Background
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+We have an issue with the "inject client cert" process where we don't always return a correct response to the client regarding the injection success. The injection is done via a WebSocket to the Kubernetes API server where we use a "tar" command and the cert content in STDIN. There is [an issue](https://github.com/kubernetes/kubernetes/issues/89899) in k8s where messages that are sent via WebSockets to Kubectl Exec are left hanging without returning a response when using STDIN. 
```

Comment:
in particular, we return a 200 OK as long as the request _should_ be processed successfully, without waiting to see if the certification injection is able to be completed

fwiw, I found the original design document and this was the intended flow for `login`:
<img width="498" alt="Screen Shot 2020-09-23 at 9 16 48 PM" src="https://user-images.githubusercontent.com/26872683/94089893-4331e100-fde2-11ea-8c8f-53132bbaa894.png">

Reply:
yes, that's correct. we may fail and not return 200 OK in cases such as Conjur host validation error or handshake error with the k8s api